pkg/entities/message.go

```diff
@@ -14,45 +14,175 @@
 
 package entities
 
-import "bytes"
+import (
+	"bytes"
+	"encoding/binary"
+	"sync"
+)
 
 const (
 	MaxTcpSocketMsgSize uint16 = 65535
 )
 
-// data struct of processed message
 type Message struct {
-	Version       uint16
-	BufferLength  uint16
-	SeqNumber     uint32
-	ObsDomainID   uint32
-	ExportTime    uint32
-	ExportAddress string
-	Set           Set
+	buffer        *bytes.Buffer
+	version       uint16
+	length        uint16
+	seqNumber     uint32
+	obsDomainID   uint32
+	exportTime    uint32
+	exportAddress string
+	isDecoding    bool
+	set           Set
+	mutex         sync.Mutex
```

Comment:
nit: Will it be better to use RWMutex to unblocking multiple reads?

Reply:
There are not that many routines doing reads in parallel with the message object. The main reason for data race is because of goroutine that does sendRefreshedTemplates in the case of UDP. In specific, message object in exporting process is shared.

Reply:
Got it. Thanks for clarifying